Keas.Mvc/Controllers/TeamAdminController.cs

```diff
@@ -371,28 +373,55 @@ public async Task<IActionResult> RemovePpsDepartment(int id, TeamPpsDepartment p
             return RedirectToAction("Index");
         }
 
+        private string GetModelErrors(ModelStateDictionary modelState)
+        {
+            var resultsList = new List<string>();
+            foreach(var result in modelState.Values)
+            {
+                foreach(var errs in result.Errors)
+                {
+                    resultsList.Add(errs.ErrorMessage);
+                }
+            }
+
+            var rtValue = "";
+            foreach(var s in resultsList)
+            {
+                rtValue = rtValue + "\n" + s;
+            }
+            return rtValue;
+        }
+
         public IActionResult Upload()
         {
-            return View();
+            var model = new List<KeyImportResults>();
+            return View(model);
         }
 
         [HttpPost]
-        public async Task<IActionResult> UploadCSV(IFormFile file)
+        public async Task<IActionResult> Upload(IFormFile file)
         {
+            var resultsView = new List<KeyImportResults>();
+
+            var userIdentity = User.Identity.Name;
+            var userName = User.GetNameClaim();

```

Comment:
@jSylvestre don't we have a helper for this somewhere?  maybe security or auth service?